File: backend/src/main/java/com/zhihire/starmap/module/interview/entity/InterviewAnswer.java
package com.zhihire.starmap.module.interview.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 用户面试回答实体
 */
@Data
@TableName("interview_answer")
public class InterviewAnswer {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long questionId;
    private String content;
    private Double aiScore;
    private String aiFeedback;
    private String matchedPoints;
    private String missedPoints;
    private LocalDateTime answeredAt;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    @TableLogic
    private String deletedAt;
}

File: backend/app/models/entities/interview_answer.py
"""Interview answer ORM entity."""
from datetime import datetime
from sqlalchemy import BigInteger, DateTime, Float, String, Text, func
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped, mapped_column
from app.models.entities.base import Base

class InterviewAnswer(Base):
    __tablename__ = "interview_answer"
    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    question_id: Mapped[int] = mapped_column(BigInteger, nullable=False, index=True)
    content: Mapped[str | None] = mapped_column(Text, nullable=True)
    ai_score: Mapped[float | None] = mapped_column(Float, nullable=True)
    ai_feedback: Mapped[str | None] = mapped_column(String(2000), nullable=True)
    matched_points: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
    missed_points: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
    answered_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=False), nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=False), nullable=False, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=False), nullable=False, server_default=func.now(), onupdate=func.now())
    deleted_at: Mapped[str] = mapped_column(String(1), nullable=False, server_default=func.text("'0'::character varying"))
    def __repr__(self) -> str:
        return f"<InterviewAnswer id={self.id} question_id={self.question_id}>"
